pkg/controller/controller_binding.go

```diff
@@ -349,38 +352,70 @@ func isPlanBindable(serviceClass *v1alpha1.ServiceClass, plan *v1alpha1.ServiceP
 
 func (c *controller) injectBinding(binding *v1alpha1.Binding, credentials map[string]interface{}) error {
 	glog.V(5).Infof("Creating Secret %v/%v", binding.Namespace, binding.Spec.SecretName)
-	secret := &v1.Secret{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      binding.Spec.SecretName,
-			Namespace: binding.Namespace,
-		},
-		Data: make(map[string][]byte),
-	}
 
+	secretData := make(map[string][]byte)
 	for k, v := range credentials {
 		var err error
-		secret.Data[k], err = serialize(v)
+		secretData[k], err = serialize(v)
 		if err != nil {
 			return fmt.Errorf("Unable to serialize credential value %q: %v; %s",
 				k, v, err)
 		}
 	}
 
-	found := false
+	secretClient := c.kubeClient.Core().Secrets(binding.Namespace)
+	// Retry loop for upserting the Secret
+	for {
```

Comment:
@pmorie Furthermore, OSB spec explicitly mentions that there's a conflict on retry:
https://github.com/openservicebrokerapi/servicebroker/blob/master/spec.md#request-4
> 409 Conflict: MUST be returned if a service binding with the same id, for the same service instance, already exists but with different parameters. The expected response body is {}, though the description field MAY be used to return a user-facing error message, as described in Broker Errors.

Well, it says "but with different parameters" but I think it's another ambiguous trap in OSB spec we need to fix, since it doesn't say that retries "with the same parameters" MUST be supported.